### PR TITLE
Fix leaky bucket bug

### PR DESCRIPTION
**Pull request by black-adder**
Fix bug where if the creditsPerSecond < itemCost, the leaky bucket never fills up. Added a maxBalance to set the upperbound of the balance.

### CODE REVIEW

File: src/samplers/ratelimiting_sampler.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default class RateLimitingSampler {
         }
 
         this._maxTracesPerSecond = maxTracesPerSecond;
-        this._rateLimiter = new RateLimiter(maxTracesPerSecond);
+        this._rateLimiter = new RateLimiter(maxTracesPerSecond, 2.0);
```
Comment (yurishkuro) [inline src/samplers/ratelimiting_sampler.js]: why `2.0`? If the `RateLimitingSampler` itself does not take this as argument, then `maxBalance` should probably be somehow calculated from `maxTracesPerSecond`

Comment (black-adder) [inline src/samplers/ratelimiting_sampler.js]: The max balance should equal the checkCredit value (1.0) in this case. But I set it to 2 so that all existing tests can work with all the edge cases which unfortunately 1.0 doesn't hit. The only difference in practice is that we buffer up to 2 traces which isn't toooooo bad.

Comment (yurishkuro) [inline src/samplers/ratelimiting_sampler.js]: but the previous behavior was `this._balance > this._creditsPerSecond`, so wouldn't it be backwards compatible to use `new RateLimiter(maxTracesPerSecond, maxTracesPerSecond)`? Otherwise if maxTracesPerSecond=10, you can never actually achieve that limit since max balance is just 2

Comment (black-adder) [inline src/samplers/ratelimiting_sampler.js]: This is supposed to not be backwards compatible. Given the new rate limit is 0.001666 (once every 10 mins), if we set maxbalance to maxtracespersecond, we'll never surpass the checkCredit value 1.0. Given that we decide what maxTracesPerSecond is, I don't think this is an issue.

File: src/rate_limiter.js [hunk 2]
```diff
@@ -22,11 +22,13 @@
 export default class RateLimiter {
     _creditsPerSecond: number;
     _balance: number;
+    _maxBalance: number;
     _lastTick: number;
 
-    constructor(creditsPerSecond: number) {
+    constructor(creditsPerSecond: number, maxBalance: number) {
         this._creditsPerSecond = creditsPerSecond;
         this._balance = creditsPerSecond;
```
Comment (yurishkuro) [inline src/rate_limiter.js]: should this be set to maxBalance instead?

File: src/samplers/ratelimiting_sampler.js (hunk 1 above)
Comment (yurishkuro) [inline src/samplers/ratelimiting_sampler.js]: I am not convinced. Currently [this test](https://github.com/uber/jaeger-client-node/blob/master/test/samplers/ratelimiting_sampler_test.js#L28) is incorrect because (a) initially it succeeds due to constructor bug(?) I pointed out, and after timer is advanced it succeeds because we're only doing a single check. If we change it to 9-10 checks, it will fail with your change.

Remember that RateLimitingSampler was originally designed for N > 1 traces per second. If you want it to work for N << 1, then let's test both scenarios.

File: test/rate_limiter.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ describe ('leaky bucket ratelimiter should', () => {
         let clock = sinon.useFakeTimers(initialDate);
```
Comment (yurishkuro) [inline test/rate_limiter.js]: I was referring to L37 when I said the test is misleading. Let's do a loop to 10 there, like in L30. Also, L31 should probably be an assert.
